File: stock/src/main/java/com/vjcspy/spring/packages/stockinfo/entity/CorEntity.java
/* (C) 2024 */
package com.vjcspy.spring.packages.stockinfo.entity;

import jakarta.persistence.*;
import java.math.BigInteger;
import java.time.LocalDate;
import lombok.*;

@Entity
@Table(name = "cor_entity")
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@NoArgsConstructor
public class CorEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "ref_id")
    private Integer refId;

    @Column(name = "cat_id")
    private Integer catId;

    @Column(name = "code", unique = true)
    private String code;

    @Column(name = "exchange")
    private String exchange;

    @Column(name = "industry_name1")
    private String industryName1;

    @Column(name = "industry_name2")
    private String industryName2;

    @Column(name = "industry_name3")
    private String industryName3;

    @Column(name = "total_shares")
    private BigInteger totalShares;

    @Column(name = "name")
    private String name;

    @Column(name = "first_trade_date")
    private LocalDate firstTradeDate;
}

File: projects/stock/src/main/kotlin/com/vjcspy/spring/packages/stockinfo/entity/CorEntity.kt
/* (C) 2024 */
package com.vjcspy.spring.packages.stockinfo.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigInteger
import java.time.LocalDate

@Entity
@Table(name = "cor_entity")
class CorEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,

    @Column(name = "\"catId\"")
    var catId: Int? = null,

    @Column(name = "\"refId\"")
    var refId: Int? = null,

    @Column(name = "code", unique = true)
    var code: String? = null,

    @Column(name = "exchange")
    var exchange: String? = null,

    @Column(name = "\"industryName1\"")
    var industryName1: String? = null,

    @Column(name = "\"industryName2\"")
    var industryName2: String? = null,

    @Column(name = "\"industryName3\"")
    var industryName3: String? = null,

    @Column(name = "\"totalShares\"")
    var totalShares: BigInteger? = null,

    @Column(name = "name")
    var name: String? = null,

    @Column(name = "\"firstTradeDate\"")
    var firstTradeDate: LocalDate? = null,
)
